enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+
+When the Metering slim installation has been enabled, the default `StorageLocation` custom resource that gets reconciled by the metering-ansible-operator should attempt to make this `CREATE DATABASE metering ...` database call using the Postgres go-client, and update it's status once that database has been created.
+
+Note: Postgresql has no native support for creating a database if it doesn't already exist, so the following call will need to be made instead:
+
+```SQL
+SELECT 'CREATE DATABASE metering' WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = 'metering')
+```
+
+#### Outlining a Potential Migration Path
+
+There's currently an undocumented PUSH API for the reporting-operator to statically inject Prometheus data that matches a specific JSON format. This API is largely used in Metering end-to-end testing scenarios and is likely, not robust or stable enough to rely on for a full data set migration from existing Metering installations. In theory, a migration effort is technically possible, but a more robust API implementation would be needed.
+
+### Risks and Mitigations
+
+- Need to support TLS and authentication for communication between the reporting-operator and the Postgres database instance.
+- Users wanting to use an existing Postgres database instance for Metering usage.
+
+## Design Details
+
+### Test Plan
+
+- TBD
+
+### Graduation Criteria
+
+- Provide a data migration path for existing Metering installations.
+- Sufficient time for feedback.
+- Sufficient testing in the following areas: upgrade, downgrade, scale, e2e.
+
+### Upgrade / Downgrade Strategy
+
+- TBD
+
+### Version Skew Strategy
+
+- N/A
+
+## Implementation History
+
+- N/A
+
+## Drawbacks
+
+- Continue to utilize the existing codebase for periodically importing metrics from Prometheus still relies on the averaged or estimated vector values that get returned by the Prometheus query_range API endpoint.
+- Deprecating the Prometheus metric importing process and working towards using the Prometheus remote_write or remote_read APIs to interact with raw metric data will require an additional migration that will be more technically difficult.
+- Extending Metering to track additional, user-provided promQL queries that are defined in the ReportDataSource custom resources will need to be done on-demand.
+- The storage and compute layers are now combined. Postgresql will now play the role as both an OLTP and OLAP type systems.
+
+## Alternatives
+
+### Dynamically Create Recording Rules
+
+Build off the ideas detailed in this proposal, but dynamically reconcile recording rules defined in the metering rule group based on promQL queries defined in a `ReportDataSource` custom resource.
```

Comment:
pros/cons of this alternative/extension?

Reply:
I updated that alternatives description. In summary, I think user-defined expensive promQL queries that could need further aggregation could be a potential pain point as Prometheus would need to load many metric samples into memory in order to process. Another problem would the additional time-series that get generated as a result of new recording rules being defined.